### PR TITLE
add endpoints to app

### PR DESCRIPTION
**Pull request by nejla-eh**

### CODE REVIEW

File: src/components/registration-view/registration-view.jsx [hunk 1]
```diff
@@ -9,19 +9,65 @@ import {
   Col,
   Row,
 } from "react-bootstrap";
+import { Link, useHistory } from "react-router-dom";
 
-//import "./registration-view.scss";
 import axios from "axios";
+import { indexOf } from "lodash";
```
Comment (0marwa0) [inline src/components/registration-view/registration-view.jsx]: I will suggest to remove this, it is probably comes from the autocomplete feature in vs code. thanks!

**Review by 0marwa0 — APPROVED** [review]
Really amazing work Nejla as always!

**Comment by 0marwa0** [comment]
Well implemented code! Thanks Nejla!

File: src/components/registration-view/registration-view.jsx (hunk 1 above)
Comment (nejla-eh) [reply src/components/registration-view/registration-view.jsx]: Thanks, just fixed it. :)

File: src/components/main-view/main-view.jsx [hunk 2]
```diff
@@ -44,75 +51,195 @@ export class MainView extends React.Component {
...
+                );
+              if (movies.length === 0) return <div className="main-view" />;
+              return (
+                <Col md={12} className="px-0">
+                  <ProfileView
+                    movies={this.state.movies}
+                    onBackClick={() => history.goBack()}
+                    onDeletedUser={() => this.onLoggedOut()}
+                    onUpdatedUser={(newUserInfo) =>
+                      this.onDeletedUser(newUserInfo)
```
Comment (0marwa0) [inline src/components/main-view/main-view.jsx]: here you should send onupdateUser funtion, thanks!

File: src/components/main-view/main-view.jsx [hunk 3]
```diff
@@ -44,75 +51,195 @@ export class MainView extends React.Component {
...
+      .catch(function (error) {
+        console.log(error);
+      });
+  }
+
+  updateUser(newUserInfo) {
+    const { userName, newPassword, email } = newUserInfo;
+    axios
+      .put(
+        `${process.env.MY_FLIX_API}/users/${currentUser}`,
```
Comment (0marwa0) [inline src/components/main-view/main-view.jsx]: currentUser is undefined.

File: src/components/main-view/main-view.jsx [hunk 4]
```diff
@@ -44,75 +51,195 @@ export class MainView extends React.Component {
...
+        {
+          Username: userName,
+          Password: newPassword,
+          Email: email,
+        },
+        {
+          headers: { Authorization: `Bearer ${token}` },
```
Comment (0marwa0) [inline src/components/main-view/main-view.jsx]: same thing for the token, you need to call it from the local storage, thanks!

File: src/components/main-view/main-view.jsx [hunk 5]
```diff
@@ -44,75 +51,195 @@ export class MainView extends React.Component {
...
+        }
+      )
+      .then((response) => {
+        console.log(response);
```
Comment (0marwa0) [inline src/components/main-view/main-view.jsx:94]: You would have to add a popup or toast when it done updating the user.

please check this example of using toast in bootstrap:
https://react-bootstrap.github.io/components/toasts/

**Review by 0marwa0 — CHANGES_REQUESTED** [review]
Please consider these changes regarding the update user function, thank you!

**Pull request merged**